ELFSage/Types/ELFHeader.lean

```diff
@@ -87,6 +90,64 @@ instance : ELFHeader ELF64Header where
   e_shnum eh      := eh.e_shnum.toNat
   e_shstrndx eh   := eh.e_shstrndx.toNat
 
+private def getHeaderClass (n: UInt8) : String := match n with
+  -- TODO: Finish populating this list of magic numbers, it is incomplete.
+  | 2 => "64-bit"
+  | _ => panic s!"Unrecognized header class {n}"
+
+private def getHeaderDataEncoding (n: UInt8) : String := match n with
+  -- TODO: Finish populating this list of magic numbers, it is incomplete.
+  | 1 => "LittleEndian"
+  | _ => panic s!"Unrecognized header data encoding {n}"
+
+private def getHeaderOS (n: UInt8) : String := match n with
+  -- TODO: Finish populating this list of magic numbers, it is incomplete.
+  | 0 => "SystemV"
+  | _ => panic s!"Unrecognized header OS/ABI {n}"
+
+private def getHeaderType (n: UInt16) : String := match n with
+  -- TODO: Finish populating this list of magic numbers, it is incomplete.
+  | 3 => "SharedObject"
+  | _ => panic s!"Unrecognized header type {n}"
+
+private def getHeaderMachine (n: UInt16) : String := match n with
+  -- TODO: Finish populating this list of magic numbers, it is incomplete.
+  | 62 => "EM_X86_64"
+  | _ => panic s!"Unrecognized header machine {n}"
+
+instance : ToString ELF64Header where toString eh :=
```

Comment:
It would also be possible to use the ELFHeader typeclass here, define
a function of the form `headerString [ELFHeader α] (eh : α) : String`, and then
define both `ToString ELF64Header` and `ToString ELF32Header` in terms of
`headerString`.

(Alternatively, it would be possible to define a [chained
instance](https://lean-lang.org/lean4/doc/typeclass.html#chaining-instances),
so that you automatically have an `ToString α` for anything that has an
`ELFHeader α` instance. 

I'm leaning against that, since 

1. I was thinking it would make sense to implement an `ELFHeader α` for the
   different ELFFile types
2. these shouldn't have the same `ToString` implementation as the ELFHeader
   types.
3. It'd be good to avoid overlapping typeclasses if possible, they can get
   confusing.